src/query/sql/src/planner/format/display_plan.rs

```diff
@@ -117,7 +119,89 @@ impl Plan {
             Plan::RefreshVirtualColumn(_) => Ok("RefreshVirtualColumn".to_string()),
 
             // Insert
-            Plan::Insert(_) => Ok("Insert".to_string()),
+            Plan::Insert(box insert) => {
+                let table_name = format!("{}.{}.{}", insert.catalog, insert.database, insert.table);
+                let inserted_columns = insert
+                    .schema
+                    .fields
+                    .iter()
+                    .map(|field| format!("{}.{} (#{})", insert.table, field.name, field.column_id))
+                    .collect::<Vec<_>>()
+                    .join(",");
+                let overwrite = insert.overwrite;
+
+                let mut children = vec![
+                    FormatTreeNode::new(format!("table: {table_name}")),
+                    FormatTreeNode::new(format!("inserted columns: [{inserted_columns}]")),
+                    FormatTreeNode::new(format!("overwrite: {overwrite}")),
+                ];
+
+                match &insert.source {
+                    InsertInputSource::SelectPlan(plan) => {
+                        if let Plan::Query {
+                            s_expr,
+                            metadata,
+                            bind_context,
+                            formatted_ast,
+                            ..
+                        } = &**plan
+                        {
+                            let metadata = &*metadata.read();
+                            let sub_tree = s_expr.to_format_tree(metadata, verbose)?;
```

Comment:
Since I moved the select plan part from `interpreter_explain.rs` to here, a not async function, it became tricky to use the old `explain_query` to get more verbose info, so I just use the `SExpr::to_format_tree`. Now it looks like this:
```
new: 
explain insert into t2 select * from t1;
----
Insert
├── table: default.default.t2
├── inserted columns: [t2.a (#0), t2.b (#1)]
├── overwrite: false
└── EvalScalar
        ├── scalars: [t1.a (#0) AS (#0), t1.b (#1) AS (#1)]
        └── Scan
                ├── table: default.t1
                ├── filters: []
                ├── order by: []
                └── limit: None
old:
explain insert into t2 select * from t1;
----
Insert
TableScan
├── table: default.default.t1
├── output columns: [a (#0), b (#1), c (#2)]
├── read rows: 0
├── read bytes: 0
├── partitions total: 0
├── partitions scanned: 0
├── push downs: [filters: [], limit: NONE]
└── estimated rows: 0.00
````